src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a contact field abstract class.
+ */
+public abstract class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
can you see that with this equals test, we can match `Address` with `Phone`. think why?